hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/AzureBlobFileSystem.java

```diff
@@ -475,6 +475,55 @@ public FileStatus getFileStatus(final Path f) throws IOException {
     }
   }
 
+  public String acquireLease(final Path f, final int duration) throws IOException {
+    LOG.debug("AzureBlobFileSystem.acquireLease path: {}", f);
+
+    Path qualifiedPath = makeQualified(f);
+
+    try {
+      return abfsStore.acquireLease(qualifiedPath, duration);
```

Comment:
how about using DurationInfo in the try with resources (logging @ debug) to track how long acquire/release took. I can imagine it can take a while to acquire. Indeed, do we have to worry about timeouts, heartbeats, etc?